planner/core/logical_plan_builder.go

```diff
@@ -4210,9 +4216,17 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if dirty || tableInfo.TempTableType == model.TempTableLocal {
 		us := LogicalUnionScan{handleCols: handleCols}.Init(b.ctx, b.getSelectOffset())
 		us.SetChildren(ds)
+		if tableInfo.Partition != nil && b.ctx.GetSessionVars().UseDynamicPartitionPrune() {
```

Comment:
I will change to check `b.optFlag&flagPartitionProcessor == 0` which is set a bit earlier in the `buildDataSource` function. Does that make more sense? Or how can I check the logical plan if it will use dynamic prune mode?